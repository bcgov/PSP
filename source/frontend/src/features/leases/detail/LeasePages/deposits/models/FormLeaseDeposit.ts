import { fromContact, IContactSearchResult, toContact } from '@/interfaces';
import { Api_SecurityDeposit } from '@/models/api/SecurityDeposit';
import { NumberFieldValue } from '@/typings/NumberFieldValue';

export class FormLeaseDeposit {
  public id?: number;
  public leaseId?: number;
  public description: string;
  public amountPaid: NumberFieldValue;
  public depositDate: string;
  public depositTypeCode: string;
  public otherTypeDescription: string;
  public contactHolder?: IContactSearchResult;
  public rowVersion: number;

  private constructor() {
    this.description = '';
    this.amountPaid = '';
    this.depositDate = '';
    this.depositTypeCode = '';
    this.otherTypeDescription = '';
    this.rowVersion = 0;
  }

  public static createEmpty(leaseId: number): FormLeaseDeposit {
    const deposit = new FormLeaseDeposit();
    deposit.leaseId = leaseId;
    return deposit;
  }

  public static fromApi(baseModel: Api_SecurityDeposit): FormLeaseDeposit {
    let model = new FormLeaseDeposit();
    model.leaseId = baseModel.leaseId ?? undefined;
    model.id = baseModel.id ?? undefined;
    model.description = baseModel.description;
    model.amountPaid = baseModel.amountPaid;
    model.depositDate = baseModel.depositDateOnly || '';
    model.depositTypeCode = baseModel.depositType.id;
    model.otherTypeDescription = baseModel.otherTypeDescription || '';
    model.contactHolder =
      baseModel.contactHolder !== null ? fromContact(baseModel.contactHolder) : undefined;

    model.rowVersion = baseModel.rowVersion || 0;
    return model;
  }

  public toApi(): Api_SecurityDeposit {
    return {
      id: this.id ?? null,
      leaseId: this.leaseId ?? null,
      description: this.description,
      amountPaid: this.amountPaid === '' ? 0 : this.amountPaid,
      depositDateOnly: this.depositDate,
      depositType: { id: this.depositTypeCode },
      otherTypeDescription: this.otherTypeDescription === '' ? null : this.otherTypeDescription,
      contactHolder: this.contactHolder !== undefined ? toContact(this.contactHolder) : null,
      depositReturns: [],
      rowVersion: this.rowVersion,
    };
  }
}
