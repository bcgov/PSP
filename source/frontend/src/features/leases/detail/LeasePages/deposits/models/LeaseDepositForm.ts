export class LeaseDepositForm {
  public returnNotes: string = '';
}
