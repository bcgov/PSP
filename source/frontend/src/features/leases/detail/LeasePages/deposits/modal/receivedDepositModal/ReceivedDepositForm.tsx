import { FastCurrencyInput, TextArea } from 'components/common/form';
import { ContactInput } from 'components/common/form/ContactInput';
import { InlineFastDatePicker, InlineInput, InlineSelect } from 'components/common/form/styles';
import { ContactManagerModal } from 'components/contact/ContactManagerModal';
import * as API from 'constants/API';
import { Formik, FormikProps } from 'formik';
import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
import { IContactSearchResult } from 'interfaces';
import * as React from 'react';
import { useState } from 'react';
import { Col, Row } from 'react-bootstrap';
import styled from 'styled-components';

import { FormLeaseDeposit } from '../../models/FormLeaseDeposit';
import { ReceivedDepositYupSchema } from './ReceivedDepositYupSchema';

export interface IReceivedDepositFormProps {
  formikRef: React.Ref<FormikProps<FormLeaseDeposit>>;
  onSave: (values: FormLeaseDeposit) => void;
  initialValues: FormLeaseDeposit;
}

export const ReceivedDepositForm: React.FunctionComponent<
  React.PropsWithChildren<IReceivedDepositFormProps>
> = ({ initialValues, formikRef, onSave }) => {
  const initialContacts =
    initialValues.contactHolder !== undefined ? [initialValues.contactHolder] : [];
  const [selectedContacts, setSelectedContacts] = useState<IContactSearchResult[]>(initialContacts);

  const [showContactManager, setShowContactManager] = useState(false);

  const lookups = useLookupCodeHelpers();
  const depositTypeOptions = lookups.getOptionsByType(API.SECURITY_DEPOSIT_TYPES);

  return (
    <Formik
      innerRef={formikRef}
      enableReinitialize
      validationSchema={ReceivedDepositYupSchema}
      onSubmit={values => {
        onSave(values);
      }}
      initialValues={initialValues}
    >
      {formikProps => (
        <StyledFormBody className="px-4">
          <Row>
            <Col md="8">
              <InlineSelect
                label="Deposit type:"
                required
                field="depositTypeCode"
                options={depositTypeOptions}
                placeholder="Select"
                onChange={() => {
                  let depositTypeCode = formikProps.values?.depositTypeCode;
                  if (!!depositTypeCode && depositTypeCode !== 'OTHER') {
                    formikProps.setFieldValue('otherTypeDescription', '');
                  }
                }}
              />
              {formikProps.values?.depositTypeCode === 'OTHER' && (
                <InlineInput label="Describe other:" field="otherTypeDescription" required />
              )}
            </Col>
          </Row>
          <Row>
            <Col>
              <TextArea label="Description:" rows={4} field="description" />
            </Col>
          </Row>
          <Row>
            <Col>
              <FastCurrencyInput
                formikProps={formikProps}
                label="Deposit Amount:"
                field="amountPaid"
                required
              />
            </Col>
            <Col>
              <InlineFastDatePicker
                formikProps={formikProps}
                label="Paid date:"
                field="depositDate"
                required
              />
            </Col>
          </Row>
          <Row>
            <Col>
              <ContactInput
                label="Deposit Holder:"
                field="contactHolder"
                setShowContactManager={setShowContactManager}
                onClear={() => {
                  formikProps.setFieldValue('contactHolder', undefined);
                  setSelectedContacts([]);
                }}
                required
              />
            </Col>
          </Row>
          <ContactManagerModal
            display={showContactManager}
            setDisplay={setShowContactManager}
            setSelectedRows={setSelectedContacts}
            selectedRows={selectedContacts}
            handleModalOk={() => {
              formikProps.setFieldValue('contactHolder', selectedContacts[0]);
              setShowContactManager(false);
            }}
            isSingleSelect
          ></ContactManagerModal>
        </StyledFormBody>
      )}
    </Formik>
  );
};

export default ReceivedDepositForm;

const StyledFormBody = styled.form`
  .form-group {
    flex-direction: column;
    .form-label {
      font-weight: bold;
    }
    textarea {
      resize: none;
    }
  }
`;
