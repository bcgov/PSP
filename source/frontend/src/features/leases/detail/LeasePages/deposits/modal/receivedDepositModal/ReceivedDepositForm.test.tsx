import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { MAX_SQL_MONEY_SIZE } from 'constants/API';
import { createMemoryHistory } from 'history';
import { mockLookups } from 'mocks/mockLookups';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { fakeText, fillInput, render, RenderOptions } from 'utils/test-utils';

import { FormLeaseDeposit } from '../../models/FormLeaseDeposit';
import ReceivedDepositForm from './ReceivedDepositForm';

const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);

const onSave = jest.fn();
const submitForm = jest.fn();

const storeState = {
  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
};

describe('ReceivedDepositForm component', () => {
  // render component under test
  const setup = (renderOptions: RenderOptions & { initialValues: FormLeaseDeposit }) => {
    const utils = render(
      <ReceivedDepositForm
        onSave={onSave}
        formikRef={{ current: { submitForm } } as any}
        initialValues={renderOptions.initialValues}
      />,
      {
        ...renderOptions,
        store: storeState,
        history,
      },
    );

    return { ...utils };
  };

  let initialValues: FormLeaseDeposit;

  beforeEach(() => {
    mockAxios.resetHistory();
    jest.clearAllMocks();
    initialValues = FormLeaseDeposit.createEmpty();
  });

  it('renders as expected', () => {
    const { asFragment } = setup({ initialValues });
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with data as expected', () => {
    const { asFragment } = setup({ initialValues });
    expect(asFragment()).toMatchSnapshot();
  });

  it('validates that the deposit date is required', async () => {
    const { container, findByDisplayValue } = setup({ initialValues });

    const { input } = await fillInput(container, 'depositDate', '2020-01-02', 'datepicker');
    await findByDisplayValue('01/02/2020');
    expect(input).toHaveProperty('required');
  });

  it('should validate required fields', async () => {
    const { container, findByText } = setup({ initialValues });

    await fillInput(container, 'depositTypeCode', '', 'select');
    expect(await findByText(/Deposit Type is required/i)).toBeVisible();

    await fillInput(container, 'description', '', 'textarea');
    expect(await findByText(/Description is required/i)).toBeVisible();

    await fillInput(container, 'amountPaid', '');
    expect(await findByText(/Deposit amount is required/i)).toBeVisible();

    await fillInput(container, 'depositDate', '');
    expect(await findByText(/Deposit Date is required/i)).toBeVisible();
  });

  it('should validate character limits', async () => {
    const { container, findByText } = setup({ initialValues });

    await fillInput(container, 'depositTypeCode', 'OTHER', 'select');
    expect(await findByText(/Describe other/i)).toBeVisible();

    await fillInput(container, 'otherTypeDescription', fakeText(201));
    expect(
      await findByText(/Other type description must be at most 200 characters/i),
    ).toBeVisible();

    await fillInput(container, 'description', fakeText(2001), 'textarea');
    expect(await findByText(/Description must be at most 2000 characters/i)).toBeVisible();

    await fillInput(container, 'amountPaid', MAX_SQL_MONEY_SIZE + 10);
    expect(await findByText(`Amount paid must be less than ${MAX_SQL_MONEY_SIZE}`)).toBeVisible();
  });
});
