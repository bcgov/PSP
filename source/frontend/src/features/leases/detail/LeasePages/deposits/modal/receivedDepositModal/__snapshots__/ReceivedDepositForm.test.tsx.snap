// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReceivedDepositForm component > renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c7.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c7.btn .Button__value {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c7.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c7.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-color: var(--surface-color-primary-button-hover);
  outline-offset: 1px;
  box-shadow: none;
}

.c7.btn.btn-primary {
  color: #FFFFFF;
  background-color: #013366;
}

.c7.btn.btn-primary:hover,
.c7.btn.btn-primary:active,
.c7.btn.btn-primary:focus {
  background-color: #1E5189;
}

.c7.btn.btn-secondary {
  color: #013366;
  background: none;
  border-color: #013366;
}

.c7.btn.btn-secondary:hover,
.c7.btn.btn-secondary:active,
.c7.btn.btn-secondary:focus {
  color: #FFFFFF;
  background-color: #013366;
}

.c7.btn.btn-info {
  color: #9F9D9C;
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c7.btn.btn-info:hover,
.c7.btn.btn-info:active,
.c7.btn.btn-info:focus {
  color: var(--surface-color-primary-button-hover);
  background: none;
}

.c7.btn.btn-light {
  color: #FFFFFF;
  background-color: #606060;
  border: none;
}

.c7.btn.btn-light:hover,
.c7.btn.btn-light:active,
.c7.btn.btn-light:focus {
  color: #FFFFFF;
  background-color: #606060;
}

.c7.btn.btn-dark {
  color: #FFFFFF;
  background-color: #474543;
  border: none;
}

.c7.btn.btn-dark:hover,
.c7.btn.btn-dark:active,
.c7.btn.btn-dark:focus {
  color: #FFFFFF;
  background-color: #474543;
}

.c7.btn.btn-danger {
  color: #FFFFFF;
  background-color: #CE3E39;
}

.c7.btn.btn-danger:hover,
.c7.btn.btn-danger:active,
.c7.btn.btn-danger:focus {
  color: #FFFFFF;
  background-color: #CE3E39;
}

.c7.btn.btn-warning {
  color: #FFFFFF;
  background-color: #FCBA19;
  border-color: #FCBA19;
}

.c7.btn.btn-warning:hover,
.c7.btn.btn-warning:active,
.c7.btn.btn-warning:focus {
  color: #FFFFFF;
  border-color: #FCBA19;
  background-color: #FCBA19;
}

.c7.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  color: var(--surface-color-primary-button-default);
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c7.btn.btn-link:hover,
.c7.btn.btn-link:active,
.c7.btn.btn-link:focus {
  color: var(--surface-color-primary-button-hover);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c7.btn.btn-link:disabled,
.c7.btn.btn-link.disabled {
  color: #9F9D9C;
  background: none;
  pointer-events: none;
}

.c7.btn:disabled,
.c7.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c7.Button .Button__icon {
  margin-right: 1.6rem;
}

.c7.Button--icon-only:focus {
  outline: none;
}

.c7.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c3 .react-datepicker__calendar-icon {
  width: 2.4rem;
  height: 3rem;
  margin-top: 0.5rem;
  margin-right: 1.2rem;
  right: 0;
  fill: var(--surface-color-primary-button-default);
  pointer-events: none;
}

.c3 .react-datepicker__view-calendar-icon input {
  padding: 0.8rem 1.2rem 0.8rem 1.2rem;
}

.c3 .react-datepicker-wrapper {
  max-width: 17rem;
}

.c5.c5.form-control.is-valid {
  background-image: none;
}

.c5.c5.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c4 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6 {
  position: relative;
  border-radius: 0.4rem;
  padding-top: 0.8rem;
  padding-bottom: 0.8rem;
  padding-left: 1.2rem;
  padding-right: 2.8rem;
  background-image: none;
  color: #9F9D9C;
  border: #606060 solid 0.1rem;
  cursor: pointer;
}

.c6.is-invalid {
  border: #CE3E39 solid 0.1rem;
}

.c2.required::before {
  content: '*';
  position: absolute;
  top: 0.75rem;
  left: 0rem;
}

.c1 {
  font-weight: bold;
}

.c0 .form-group {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .form-group .form-label {
  font-weight: bold;
}

.c0 .form-group textarea {
  resize: none;
}

<form
    class="c0"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="pb-2 row"
        >
          <div
            class="pr-0 text-left col-4"
          >
            <label
              class="c1"
            >
              Deposit type:
            </label>
          </div>
          <div
            class="c2 required text-left col-5"
          >
            <div
              class="form-group"
            >
              <select
                class="form-select form-control"
                id="input-depositTypeCode"
                name="depositTypeCode"
              >
                <option
                  value=""
                >
                  Select...
                </option>
                <option
                  class="option"
                  data-testid="select-option-OTHER"
                  value="OTHER"
                >
                  Other deposit
                </option>
                <option
                  class="option"
                  data-testid="select-option-PET"
                  value="PET"
                >
                  Pet deposit
                </option>
                <option
                  class="option"
                  data-testid="select-option-SECURITY"
                  value="SECURITY"
                >
                  Security deposit
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="textarea input form-group"
        >
          <label
            class="form-label"
            for="input-description"
          >
            Description: 
          </label>
          <textarea
            class="form-control"
            id="input-description"
            name="description"
            rows="4"
            title=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="required form-group"
        >
          <label
            class="form-label"
            for="input-amountPaid"
          >
            Deposit amount: 
          </label>
          <div
            class="input-tooltip-wrapper"
          >
            <input
              class="form-control input-number"
              id="input-amountPaid"
              inputmode="numeric"
              name="amountPaid"
              placeholder=""
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="c3 required c4 form-group"
        >
          <label
            class="form-label"
            for="datepicker-depositDate"
          >
            <span
              class="datepicker-label"
            >
              Paid date:
            </span>
             
          </label>
          <div
            class="react-datepicker-wrapper d-block"
          >
            <div
              class="react-datepicker__input-container react-datepicker__view-calendar-icon"
            >
              <svg
                class="react-datepicker__calendar-icon"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                />
              </svg>
              <input
                autocomplete="off"
                class="c5 form-control date-picker"
                id="datepicker-depositDate"
                name="depositDate"
                placeholder="MMM DD, YYYY"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-9"
      >
        <div
          class="required input form-group"
        >
          <label
            class="form-label"
            for="input-contactHolder"
          >
            Deposit holder:
          </label>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="c6"
              >
                Select from contacts
              </div>
              <div
                class="d-none input form-group"
              >
                <input
                  class="form-control"
                  id="input-contactHolder.id"
                  name="contactHolder.id"
                  placeholder="Select from Contacts"
                  value=""
                />
              </div>
            </div>
            <div
              class="pl-0 col-auto"
            >
              <button
                class="c7 Button Button--icon-only px-2 btn btn-primary"
                type="button"
              >
                <div
                  class="Button__icon"
                >
                  <svg
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-12"
      >
        <div
          style="margin-top: 24px;"
        >
          <p>
            Do you want to save it?
          </p>
        </div>
      </div>
    </div>
    <div />
  </form>
</DocumentFragment>
`;

exports[`ReceivedDepositForm component > renders with data as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c7.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c7.btn .Button__value {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c7.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c7.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-color: var(--surface-color-primary-button-hover);
  outline-offset: 1px;
  box-shadow: none;
}

.c7.btn.btn-primary {
  color: #FFFFFF;
  background-color: #013366;
}

.c7.btn.btn-primary:hover,
.c7.btn.btn-primary:active,
.c7.btn.btn-primary:focus {
  background-color: #1E5189;
}

.c7.btn.btn-secondary {
  color: #013366;
  background: none;
  border-color: #013366;
}

.c7.btn.btn-secondary:hover,
.c7.btn.btn-secondary:active,
.c7.btn.btn-secondary:focus {
  color: #FFFFFF;
  background-color: #013366;
}

.c7.btn.btn-info {
  color: #9F9D9C;
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c7.btn.btn-info:hover,
.c7.btn.btn-info:active,
.c7.btn.btn-info:focus {
  color: var(--surface-color-primary-button-hover);
  background: none;
}

.c7.btn.btn-light {
  color: #FFFFFF;
  background-color: #606060;
  border: none;
}

.c7.btn.btn-light:hover,
.c7.btn.btn-light:active,
.c7.btn.btn-light:focus {
  color: #FFFFFF;
  background-color: #606060;
}

.c7.btn.btn-dark {
  color: #FFFFFF;
  background-color: #474543;
  border: none;
}

.c7.btn.btn-dark:hover,
.c7.btn.btn-dark:active,
.c7.btn.btn-dark:focus {
  color: #FFFFFF;
  background-color: #474543;
}

.c7.btn.btn-danger {
  color: #FFFFFF;
  background-color: #CE3E39;
}

.c7.btn.btn-danger:hover,
.c7.btn.btn-danger:active,
.c7.btn.btn-danger:focus {
  color: #FFFFFF;
  background-color: #CE3E39;
}

.c7.btn.btn-warning {
  color: #FFFFFF;
  background-color: #FCBA19;
  border-color: #FCBA19;
}

.c7.btn.btn-warning:hover,
.c7.btn.btn-warning:active,
.c7.btn.btn-warning:focus {
  color: #FFFFFF;
  border-color: #FCBA19;
  background-color: #FCBA19;
}

.c7.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  color: var(--surface-color-primary-button-default);
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c7.btn.btn-link:hover,
.c7.btn.btn-link:active,
.c7.btn.btn-link:focus {
  color: var(--surface-color-primary-button-hover);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c7.btn.btn-link:disabled,
.c7.btn.btn-link.disabled {
  color: #9F9D9C;
  background: none;
  pointer-events: none;
}

.c7.btn:disabled,
.c7.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c7.Button .Button__icon {
  margin-right: 1.6rem;
}

.c7.Button--icon-only:focus {
  outline: none;
}

.c7.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c3 .react-datepicker__calendar-icon {
  width: 2.4rem;
  height: 3rem;
  margin-top: 0.5rem;
  margin-right: 1.2rem;
  right: 0;
  fill: var(--surface-color-primary-button-default);
  pointer-events: none;
}

.c3 .react-datepicker__view-calendar-icon input {
  padding: 0.8rem 1.2rem 0.8rem 1.2rem;
}

.c3 .react-datepicker-wrapper {
  max-width: 17rem;
}

.c5.c5.form-control.is-valid {
  background-image: none;
}

.c5.c5.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c4 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6 {
  position: relative;
  border-radius: 0.4rem;
  padding-top: 0.8rem;
  padding-bottom: 0.8rem;
  padding-left: 1.2rem;
  padding-right: 2.8rem;
  background-image: none;
  color: #9F9D9C;
  border: #606060 solid 0.1rem;
  cursor: pointer;
}

.c6.is-invalid {
  border: #CE3E39 solid 0.1rem;
}

.c2.required::before {
  content: '*';
  position: absolute;
  top: 0.75rem;
  left: 0rem;
}

.c1 {
  font-weight: bold;
}

.c0 .form-group {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .form-group .form-label {
  font-weight: bold;
}

.c0 .form-group textarea {
  resize: none;
}

<form
    class="c0"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="pb-2 row"
        >
          <div
            class="pr-0 text-left col-4"
          >
            <label
              class="c1"
            >
              Deposit type:
            </label>
          </div>
          <div
            class="c2 required text-left col-5"
          >
            <div
              class="form-group"
            >
              <select
                class="form-select form-control"
                id="input-depositTypeCode"
                name="depositTypeCode"
              >
                <option
                  value=""
                >
                  Select...
                </option>
                <option
                  class="option"
                  data-testid="select-option-OTHER"
                  value="OTHER"
                >
                  Other deposit
                </option>
                <option
                  class="option"
                  data-testid="select-option-PET"
                  value="PET"
                >
                  Pet deposit
                </option>
                <option
                  class="option"
                  data-testid="select-option-SECURITY"
                  value="SECURITY"
                >
                  Security deposit
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="textarea input form-group"
        >
          <label
            class="form-label"
            for="input-description"
          >
            Description: 
          </label>
          <textarea
            class="form-control"
            id="input-description"
            name="description"
            rows="4"
            title=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="required form-group"
        >
          <label
            class="form-label"
            for="input-amountPaid"
          >
            Deposit amount: 
          </label>
          <div
            class="input-tooltip-wrapper"
          >
            <input
              class="form-control input-number"
              id="input-amountPaid"
              inputmode="numeric"
              name="amountPaid"
              placeholder=""
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="c3 required c4 form-group"
        >
          <label
            class="form-label"
            for="datepicker-depositDate"
          >
            <span
              class="datepicker-label"
            >
              Paid date:
            </span>
             
          </label>
          <div
            class="react-datepicker-wrapper d-block"
          >
            <div
              class="react-datepicker__input-container react-datepicker__view-calendar-icon"
            >
              <svg
                class="react-datepicker__calendar-icon"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                />
              </svg>
              <input
                autocomplete="off"
                class="c5 form-control date-picker"
                id="datepicker-depositDate"
                name="depositDate"
                placeholder="MMM DD, YYYY"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-9"
      >
        <div
          class="required input form-group"
        >
          <label
            class="form-label"
            for="input-contactHolder"
          >
            Deposit holder:
          </label>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="c6"
              >
                Select from contacts
              </div>
              <div
                class="d-none input form-group"
              >
                <input
                  class="form-control"
                  id="input-contactHolder.id"
                  name="contactHolder.id"
                  placeholder="Select from Contacts"
                  value=""
                />
              </div>
            </div>
            <div
              class="pl-0 col-auto"
            >
              <button
                class="c7 Button Button--icon-only px-2 btn btn-primary"
                type="button"
              >
                <div
                  class="Button__icon"
                >
                  <svg
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-12"
      >
        <div
          style="margin-top: 24px;"
        >
          <p>
            Do you want to save it?
          </p>
        </div>
      </div>
    </div>
    <div />
  </form>
</DocumentFragment>
`;
